modules/sfp_dehashed.py

```diff
@@ -0,0 +1,213 @@
+            'favIcon': "https://www.dehashed.com/assets/img/favicon.ico",
+            'logo': "https://www.dehashed.com/assets/img/logo.png",
+            'description': "Have you been compromised? "
+            "DeHashed provides free deep-web scans and protection against credential leaks. "
+            "A modern personal asset search engine created for "
+            "security analysts, journalists, security companies, "
+            "and everyday people to help secure accounts and provide insight on compromised assets. "
+            "Free breach alerts & breach notifications.",
+        }
+    }
+
+    # Default options
+    opts = {
+        'api_key_username': '',
+        'api_key': '',
+        'per_page': 10000,
+        'max_pages': 2,
+        'pause': 1
+    }
+
+    # Option descriptions
+    optdescs = {
+        'api_key_username': 'Username for api key',
+        'api_key': 'Token for api key',
+        'per_page': 'Maximum number of results per page.(Max: 10000)',
+        'max_pages': 'Maximum number of pages to fetch(Max: 10 pages)',
+        'pause': 'Amount of time to wait before each API call'
+    }
+
+    results = None
+    errorState = False
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return [
+            "DOMAIN_NAME",
+            "EMAILADDR"
+        ]
+
+    # What events this module produces
+    def producedEvents(self):
+        return [
+            'EMAILADDR',
+            'EMAILADDR_COMPROMISED',
+            'PASSWORD_COMPROMISED',
+            'HASH_COMPROMISED',
+            'RAW_RIR_DATA'
+        ]
+
+    # Query Dehashed
+    def query(self, event, per_page, start):
+        if event.eventType == "EMAILADDR":
+            queryString = f"https://api.dehashed.com/search?query=email:\"{event.data}\"&page={start}&size={self.opts['per_page']}"
+        if event.eventType == "DOMAIN_NAME":
+            queryString = f"https://api.dehashed.com/search?query=email:\"@{event.data}\"&page={start}&size={self.opts['per_page']}"
+
+        token = (base64.b64encode(self.opts['api_key_username'].encode('utf8') + ":".encode('utf-8') + self.opts['api_key'].encode('utf-8'))).decode('utf-8')
+        headers = {
+            'Accept': 'application/json',
+            'Authorization': f'Basic {token}'
+        }
+
+        res = self.sf.fetchUrl(queryString,
+                               headers=headers,
+                               timeout=15,
+                               useragent=self.opts['_useragent'],
+                               verify=False)
+
+        time.sleep(self.opts['pause'])
+
+        if res['code'] == "400":
+            self.sf.error("Too many requests were performed in a small amount of time. Please wait a bit before querying the API.")
```

Comment:
Not sure how useful such an error would be since it would occur mid-scan, and the SpiderFoot user wouldn't be able to do anything about it. Better if it backed off and tried again after a few seconds.